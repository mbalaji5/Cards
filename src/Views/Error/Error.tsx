interface propType {
  errorMessage: string;
}
const Error = ({ errorMessage }: propType) => {
  return (
    <>
      <label id="error">{errorMessage}</label>
    </>
  );
};

Error.defaultProps = { errorMessage: "error" };

export default Error;
