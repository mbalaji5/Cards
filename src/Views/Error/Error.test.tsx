import { render } from "@testing-library/react";
import React from "react";

import Error from "./Error";
const errorMessage = "Sorry you are not eligible for neither C1 nor C2.";
describe("Should Validate Error Component", () => {
  it("Should validate default param", () => {
    const { container } = render(<Error />);
    expect(container.textContent).toBe("error");
  });
  it("Should validate props", () => {
    const { container } = render(<Error errorMessage={errorMessage} />);
    expect(container.textContent).toBe(errorMessage);
  });
  it("should match snopshot", () => {
    const { baseElement } = render(<Error />);
    expect(baseElement).toMatchSnapshot();
  });
  it("should match snopshot", () => {
    const { baseElement } = render(<Error errorMessage={errorMessage} />);
    expect(baseElement).toMatchSnapshot();
  });
});
