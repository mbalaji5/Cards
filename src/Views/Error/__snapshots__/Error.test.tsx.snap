// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Validate Error Component should match snopshot 1`] = `
<body>
  <div>
    <label
      id="error"
    >
      error
    </label>
  </div>
</body>
`;

exports[`Should Validate Error Component should match snopshot 2`] = `
<body>
  <div>
    <label
      id="error"
    >
      Sorry you are not eligible for neither C1 nor C2.
    </label>
  </div>
</body>
`;
